### PR TITLE
User/Rajivts/UnitTestCasesForSystemXml

### PR DESCRIPTION
**Pull request by Rajivts**
Initial Test Case creation for System.Xml.XmlTextReader

### CODE REVIEW

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class XmlTextReaderAttributeTests
+    {
+        [Fact]
+        public void XmlTextReaderGetAttributeWithIndexTest()
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: I think this should be couple of more test cases:
- negative index
- 0 (as is), 1 (add 1 more attribute), 2 (out of bounds)

Comment (krwq) [reply src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: also you can remove the prefix "XmlTextReader" from the test case name as it is already contained in the class name

Comment (krwq) [reply src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: also one more test case where value is not there at all

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val'> abc </element1>");            
+            Assert.True(textReader.Read());
+            Assert.Equal("val", textReader.GetAttribute(0));
+        }
+
+        [Fact]
+        public void XmlTextReaderMoveToAttributeWithNameTest()
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: please add another more test cases where: attribute is not there, there is more than one attribute with the same name (check if moved to the correct one)

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val'> abc </element1>");
+            Assert.True(textReader.Read());
+            Assert.True(textReader.MoveToAttribute("attr"));
+            Assert.Equal(XmlNodeType.Attribute, textReader.NodeType);
+            Assert.Equal("val", textReader.Value);
+        }
+
+        [Fact]
+        public void XmlTextReaderMoveToAttributeWithNamespace()
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: - two namespaces and 1 attribute in each namespace - check if chosen the correct one
- why is the NameTable argument needed here?

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderConstructionTests.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+using Xunit;
+using System.IO;
+
+namespace System.Xml.Tests
+{
+    public class XmlTextReaderConstructionTests
+    {
+        [Fact]
+        public void TextReaderConstructionWithStream()
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderConstructionTests.cs]: I'd go with something like`CtorNullBaseUriStreamNameTable` or just `CtorNullBaseUri`

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+using System.Text;
+
+namespace System.Xml.Tests
+{
+    public class XmlTextReaderPropertiesTests
+    {
+        [Fact]
+        public void XmlTextReaderHasValueTest()
+        {
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val'> abc </element1>");
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs]: I'd unify some of those tests into one larger test cases with multiple asserts where possible

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderTestHelper.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+{
+    internal static class XmlTextReaderTestHelper
+    {
+        internal static XmlTextReader CreateReader(string input)
+        {
+            var textReader = new XmlTextReader(input, XmlNodeType.Element, null);
+            return textReader;
+        }
+
+        internal static XmlTextReader CreateReader(string input, XmlNameTable nameTable)
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderTestHelper.cs]: I haven't seen where the nameTable was really used in the testcase. possibly we could remove it?

**Comment by Rajivts** [comment]
@krwq Updated the PR with changes

**Comment by Rajivts** [comment]
Removed the test based on BaseURI for XmlTextReader, since providing a non-null baseURI results in an exception preventing the tests from running. Also, fixed a couple more tests

**Comment by Rajivts** [comment]
@krwq Made the relevant changes, PR is ready for review

**Comment by krwq** [comment]
@Rajivts what error are you seeing for non-null baseURI? Is it perhaps a bug in XML?

File: src/System.Private.Xml/System.Private.Xml.sln [hunk 7]
```diff
@@ -158,136 +158,272 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{1A2F9F4A
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{E107E9C1-E893-4E87-987E-04EF0DCEAEFD}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Xml.XmlTextReader.Tests", "tests\XmlTextReader\System.Xml.XmlTextReader.Tests\System.Xml.XmlTextReader.Tests.csproj", "{7FDF3054-AF3D-4804-88AE-37A9D62E4E3B}"
```
Comment (krwq) [inline src/System.Private.Xml/System.Private.Xml.sln]: please put the project under `tests\Writers`

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs [hunk 8]
```diff
@@ -9,33 +9,47 @@ namespace System.Xml.Tests
...
 
         [Fact]
-        public void XmlTextReaderMoveToAttributeWithNameTest()
+        public void MoveToAttributeWithNameTest()
         {
-            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val'> abc </element1>");
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val' attr2='val2'> abc </element1>");
             Assert.True(textReader.Read());
-            Assert.True(textReader.MoveToAttribute("attr"));
+            Assert.True(textReader.MoveToAttribute("attr"));
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: extra spaces in the end

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs [hunk 9]
```diff
@@ -9,33 +9,47 @@ namespace System.Xml.Tests
...
+            Assert.False(textReader.MoveToAttribute("attr2"));
         }
 
         [Fact]
-        public void XmlTextReaderMoveToAttributeWithNamespace()
+        public void MoveToAttributeWithNamespace()
         {
-            XmlTextReader textReader = 
-                XmlTextReaderTestHelper.CreateReader("<List xmlns:tp='urn:NameSpace'><element1 tp:attr='val'>abc</element1></List>", new NameTable());
+            string inp = "<List xmlns:tp='urn:NameSpace'><element1 tp:attr='val' /><element1 xmlns:rp='urn' rp:attr='val2' /></List>";
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs]: `inp` -> `xml`

File: src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs [hunk 10]
```diff
@@ -10,7 +10,7 @@ namespace System.Xml.Tests
     public class XmlTextReaderPropertiesTests
     {
         [Fact]
-        public void XmlTextReaderHasValueTest()
+        public void HasValueTest()
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs]: please unify the test cases with the same input in this class

Comment (krwq) [reply src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs]: (only for test cases where there is only ctor and property value check)

File: src/System.Private.Xml/System.Private.Xml.sln (hunk 7 above)
Comment (krwq) [reply src/System.Private.Xml/System.Private.Xml.sln]: yes 🤦‍♂️

**Comment by krwq** [comment]
Hey @Rajivts, do you think you will find some time to update this PR with the requested changes?

**Comment by karelz** [comment]
@Rajivts ping?

**Comment by karelz** [comment]
Closing for now. @Rajivts feel free to reopen once you have time. Thanks!

**Pull request closed without merging**